src/encrypt.rs

```diff
@@ -0,0 +1,301 @@
+//! COSE Encryption
+
+use serde::{Deserialize, Serialize, Serializer, ser::SerializeSeq};
+use serde_bytes::ByteBuf;
+use serde_cbor::Error as CborError;
+use serde_cbor::Value as CborValue;
+use openssl::symm::{encrypt_aead, decrypt_aead, Cipher};
+use openssl::rand::rand_bytes;
+
+use crate::error::COSEError;
+use crate::header_map::{HeaderMap, map_to_empty_or_serialized};
+
+///
+pub enum COSEAlgorithm {
+    /// AES-GCM mode w/ 128-bit key, 128-bit tag
+    A128GCM,
+    /// AES-GCM mode w/ 192-bit key, 128-bit tag
+    A192GCM,
+    /// AES-GCM mode w/ 256-bit key, 128-bit tag
+    A256GCM,
+    /// AES-CCM mode 128-bit key, 64-bit tag, 13-byte nonce
+    AESCCM1664128,
+    /// AES-CCM mode 256-bit key, 64-bit tag, 13-byte nonce
+    AESCCM1664256,
+    /// AES-CCM mode 128-bit key, 64-bit tag, 7-byte nonce
+    AESCCM6464128,
+    /// AES-CCM mode 256-bit key, 64-bit tag, 13-byte nonce
+    AESCCM6464256,
+    /// AES-CCM mode 128-bit key, 128-bit tag, 13-byte nonce
+    AESCCM16128128,
+    /// AES-CCM mode 256-bit key, 128-bit tag, 13-byte nonce
+    AESCCM16128256,
+    /// AES-CCM mode 128-bit key, 128-bit tag, 7-byte nonce
+    AESCCM64128128,
+    /// AES-CCM mode 128-bit key, 128-bit tag, 7-byte nonce
+    AESCCM64128256
+}
+
+impl COSEAlgorithm {
+    ///
+    pub fn value(&self) -> usize {
```

Comment:
I think a bunch of functions in this `impl` are public where they don't need to be.
I can't see a reason to have them `pub` unless needed by someone.